lib/entry.js

```diff
@@ -3,14 +3,14 @@
 var DIRECTORY_MODE = 16877;
 
 module.exports = Entry;
-function Entry(relativePath, size, mtime) {
+function Entry(relativePath, size, mtime, mode) {
   var isDirectory = relativePath.charAt(relativePath.length - 1) === '/';
 
    // ----------------------------------------------------------------------
    // required properties
 
   this.relativePath = relativePath;
-  this.mode = isDirectory ? DIRECTORY_MODE : 0;
+  this.mode = mode || (isDirectory ? DIRECTORY_MODE : 0);
```

Comment:
I think we should do this based on arity not based on truthiness. What if the mode we want is `0`.

```js
if  (arguments.length > 3) {
  this.mode = mode;
} else {
   var isDirectory = relativePath.charAt(relativePath.length - 1) === '/';
   this.mode = (isDirectory ? DIRECTORY_MODE : 0);
}
```